docs/getting-started/advanced-guide/schema.md

```diff
@@ -0,0 +1,429 @@
+Groups must be completely unique and cannot be shared between edges and entities.
+
+### Edges
+Edges must have the following:
+
+- `source` - The type of object that will be used as the source vertex in your graph. Needs to be a type defined in the [Types schema](#types).
+- `destination` - Similar to source, it can either be the same or a different type.
+- `directed` - Tells Gaffer if the edge is directed or undirected. Needs to be a type which means true or false, see [Types](#true--false) for more info.
+
+When an Edge is undirected in Gaffer (`directed` is `false`), it is treated as if the relationship was bidirectional and the vertices of the edge do not have an authoritative source and destination thus, the undirected edges `A -- B` and `B -- A` are equal, and therefore will be aggregated together.
+Gaffer will present the undirected edges Vertices in natural ordering, so a client will also see the above edge presented as `A, B`.
+It will be aggregated with any other edge that has the same source and destination and is also undirected.
+For example an undirected Edge that is added, `A -> B`, will be aggregated with another Edge of `B -> A` if they are undirected.
+
+### Entities
+Entities must have a `vertex` field, which is similar to the `source` and `destination` fields on an edge.
+For example, modelling `London -> Paris` requires an entity definition to represent the city and an edge definition to represent the relationship.
+In that example the entity `vertex` field and the edge `source` and `destination` fields would all be of the same type.
+
+### Optional Element fields
+Edges and Entities can optionally have the following fields:
+
+- `description` - A simple string which should provide some context and explain what the element is.
+- `parents` - An array of parent group names. These must relate to the same sort of element as the child, for example an edge cannot have an entity as a parent. Elements can inherit any information from multiple parent elements. Fields will be merged/overridden, so the hierarchy of parents is important. Any fields that are defined in the child element will also merge or override information taken from the parents.
+- `properties` - Properties are defined by a map of key-value pairs of property names to property types. Property types are described in the Types schema.
+- `groupBy` - By default Gaffer uses the element group and its vertices to group similar elements together when aggregating and summarising elements. This groupBy field allows you to specify extra properties (in addition to the element group and vertices) to use for controlling when similar elements should be grouped together and summarised.
+- `visibilityProperty` - When using visibility properties in your graph, ensure the sensitive elements have a visibility property and then set this visibilityProperty field to that property name. This ensures Gaffer knows to restrict access to sensitive elements.
```

Comment:
I would potentially swap the sentence order, you usually would want the quick overview of what the property is before the explanation of how to implement it.

Reply:
I've reworded/reordered this. Should now be clearer what each property is first, before going into more detailed.